pvfactors/geometry/pvarray.py

```diff
@@ -130,6 +130,7 @@ def cast_shadows(self):
         """Use calculated solar_2d_vector and array configuration to calculate
         shadows being casted in the ordered pv array.
         The logic here is quite specific to ordered pv arrays"""
+        tol=1e-6
```

Comment:
Does it work when you use the ``DISTANCE_TOLERANCE`` defined here? https://github.com/SunPower/pvfactors/blob/master/pvfactors/config.py#L31
If not, could you please move this new tolerance variable to the ``config.py`` module and write it in UPPERCASE?

I like to keep track of all these factors and simulation constants in the ``config.py`` module.

Reply:
Minimal example to reproduce error.
The global `DISTANCE_TOLERANCE` works.
```
from pvfactors.geometry.pvarray import OrderedPVArray
params = {'axis_azimuth': 0, 
          'gcr': 0.3,
          'n_pvrows': 3,
          'pvrow_height': 1.8,
          'pvrow_width': 1.98,
          'solar_azimuth': 263.99310644558074,
          'solar_zenith': 73.91658668648401, 
          'surface_azimuth': 270.0, 
          'surface_tilt': 51.98206680806641}
pvarray_w_direct_shading = OrderedPVArray.from_dict(params)
pvarray_w_direct_shading.cast_shadows()
```

Reply:
I added this as a test in geometry, it is very picky about the values, what I understand is that it is casting a super small shadow over the next row and somehow it gets lost after that.

Reply:
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-2-45dfcda9d7ec> in <module>
     10           'surface_tilt': 51.98206680806641}
     11 pvarray_w_direct_shading = OrderedPVArray.from_dict(params)
---> 12 pvarray_w_direct_shading.cast_shadows()

c:\apps\pvfactors\pvfactors\geometry\pvarray.py in cast_shadows(self)
    183             # Use distance translation to cast shadows on pvrows
    184             for idx, pvrow in enumerate(list_pvrows_to_shade):
--> 185                 proj = Point(proj_initial.x + idx * self.distance,
    186                              proj_initial.y)
    187                 shaded_side = getattr(pvrow, self.illum_side)

AttributeError: 'GeometryCollection' object has no attribute 'x'
```

Reply:
Thank you @tcapelle, do you get that error running the test that you implemented AND after removing the added tolerance in ``cast_shadows()``?

Reply:
yeap